projects/driver_controls/src/fsm/pedal_calibration_fsm.c

```diff
@@ -0,0 +1,116 @@
+#include "pedal_calibration_fsm.h"
+#include "input_event.h"
+#include "log.h"
+
+FSM_DECLARE_STATE(state_start);
+FSM_DECLARE_STATE(state_full_brake);
+FSM_DECLARE_STATE(state_full_throttle);
+FSM_DECLARE_STATE(state_enter_validation);
+FSM_DECLARE_STATE(state_continue_validation);
+
+FSM_STATE_TRANSITION(state_start) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_FULL_BRAKE, state_full_brake);
+
+  // User can go to validation stage if already calibrated.
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_ENTER_VALIDATION, state_enter_validation);
+}
+
+FSM_STATE_TRANSITION(state_full_brake) {
+  // The only available next step is full throttle state.
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_FULL_THROTTLE, state_full_throttle);
+
+  // Or repeating full brake stage if pedal was not in the correct position at some point.
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_FULL_BRAKE, state_full_brake);
+}
+
+FSM_STATE_TRANSITION(state_full_throttle) {
+  // The only available next step is validate state. The user must validate the calibration.
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_ENTER_VALIDATION, state_enter_validation);
+
+  // Or repeating full throttle stage if pedal was not in the correct position at some point.
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_FULL_THROTTLE, state_full_throttle);
+}
+
+FSM_STATE_TRANSITION(state_enter_validation) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_CONTINUE_VALIDATION, state_continue_validation);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_START, state_start);
+}
+
+FSM_STATE_TRANSITION(state_continue_validation) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_CONTINUE_VALIDATION, state_continue_validation);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_CALIBRATION_START, state_start);
+}
+
+// Output function for the start state.
+static void prv_start_output(FSM *fsm, const Event *e, void *context) {
+  LOG_DEBUG("You are at the menu of pedal calibration.\n");
+}
+
+// Output function for the full brake state.
+// Reads and collects pedal inputs assuming pedal is in full brake position.
+static void prv_full_brake_output(FSM *fsm, const Event *e, void *context) {
+  LOG_DEBUG("Keep the pedal at full brake position.\n");
+  PedalCalibrationStorage *storage = context;
+  pedal_calibration_process_state(storage, PEDAL_CALIBRATION_STATE_FULL_BRAKE);
```

Comment:
Might be worth adding a short delay here to ensure the user is in position?

Reply:
Yeah the process itself is kinda hacked together - I'll probably redo the process-related stuff.